tests/queries/0_stateless/02918_parallel_replicas_custom_key_unavailable_replica.sql

```diff
@@ -0,0 +1,33 @@
+DROP TABLE IF EXISTS 02918_parallel_replicas;
+
+CREATE TABLE 02918_parallel_replicas (x String, y Int32) ENGINE = MergeTree ORDER BY cityHash64(x);
+
+INSERT INTO 02918_parallel_replicas SELECT toString(number), number % 4 FROM numbers(1000);
+
+-- SET async_socket_for_remote=0;
+-- SET async_query_sending_for_remote=0;
+-- SET load_balancing='round_robin';
+SET prefer_localhost_replica=0;
+
+-- { echoOn }
+SELECT y, count()
```

Comment:
What will happen if we say `max_parallel_replicas=4`? Will the query be `cityHash64(y)/3` or will it be `cityHash64(y)/4` and the missing replica will be reassigned?

Reply:
It will be `cityHash64(y)/4` and data partitioned by custom key should be reassigned among available replicas in round-robin fashion

Reply:
That seems wrong to me. It will lead to one replica doing 2x more work.

Can't we instead adapt the condition to  the number of replicas available and disable custom parallel replicas if there is only one?

Reply:
Now that I think of it, we probably don't want to do that since it involves waiting for a dead/slow replica (which should be the odd case).

Let's just add a test to verify the behaviour and result is correct now